src/main/java/Parser.java

```diff
@@ -0,0 +1,54 @@
+import java.time.DateTimeException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Parser {
+    public Parser() {
+
+    }
+    private static ArrayList<String> tokenize(String input) {
+        ArrayList<String> tokens = new ArrayList<>(Arrays.asList(input.split(" ")));
+        tokens.removeIf(s -> s.equals(" ") || s.equals(""));
+        tokens.forEach(s -> s = s.trim());
+        //System.out.println(tokens);
+        return tokens;
+    }
+    public static Command parse(String userIn) throws DateTimeException, DukeException {
+        ArrayList<String> tokens = tokenize(userIn);
+        String key = tokens.get(0);
+        Command ret = null;
+
+        switch (key) {
+            case "bye":
```

Comment:
According to the java styles, the switch and case should be in the same indentation.